contracts/mainnet/connectors/morpho-blue/main.sol

```diff
@@ -0,0 +1,918 @@
+			_amt
+		);
+
+		uint256 _shares;
+		(_assets, _shares) = MORPHO_BLUE.supply(
+			_marketParams,
+			_amt,
+			0,
+			address(this),
+			new bytes(0)
+		);
+
+		setUint(_setId, _assets);
+
+		_eventName = "LogSupplyAssets((address,address,address,address,unit256),unit256,unit256,unit256,unit256)";
+		_eventParam = abi.encode(
+			_marketParams,
+			_assets,
+			_shares,
+			_getId,
+			_setId
+		);
+	}
+
+	/**
+	 * @dev Supply ETH/ERC20 Token for lending.
+	 * @notice Supplies assets to Morpho Blue for lending.
+	 * @param _marketParams The market to supply assets to. (For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param _assets The amount of assets to supply. (For max: `uint256(-1)`)
+	 * @param _onBehalf The address that will get the shares.
+	 * @param _getId ID to retrieve amt.
+	 * @param _setId ID stores the amount of tokens deposited.
+	 */
+	function supplyOnBehalf(
+		MarketParams memory _marketParams,
+		uint256 _assets,
+		address _onBehalf,
+		uint256 _getId,
+		uint256 _setId
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		uint256 _amt;
+		(
+			_marketParams, // Updated token contracts in case of Eth
+			_amt
+		) = _performEthToWethConversion(
+			_marketParams,
+			_assets,
+			_onBehalf,
+			_getId,
+			Mode.Other
+		);
+
+		approve(
+			TokenInterface(_marketParams.loanToken),
+			address(MORPHO_BLUE),
+			_amt
+		);
+
+		uint256 _shares;
+		(_assets, _shares) = MORPHO_BLUE.supply(
+			_marketParams,
+			_amt,
+			0,
+			_onBehalf,
+			new bytes(0)
+		);
+
+		setUint(_setId, _assets);
+
+		_eventName = "LogSupplyOnBehalf((address,address,address,address,unit256),uint256,uint256,address,uint256,uint256)";
+		_eventParam = abi.encode(
+			_marketParams,
+			_assets,
+			_shares,
+			_onBehalf,
+			_getId,
+			_setId
+		);
+	}
+
+	/**
+	 * @dev Supply ETH/ERC20 Token for lending.
+	 * @notice Supplies assets for a perfect share amount to Morpho Blue for lending.
+	 * @param _marketParams The market to supply assets to. (For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param _shares The exact amount of shares to mint. (For max: `uint256(-1)`)
+	 * @param _onBehalf The address that will get the shares.
+	 * @param _getId ID to retrieve amt.
+	 * @param _setId ID stores the amount of tokens deposited.
+	 */
+	function supplySharesOnBehalf(
+		MarketParams memory _marketParams,
+		uint256 _shares,
+		address _onBehalf,
+		uint256 _getId,
+		uint256 _setId
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		uint256 _amt;
+		(
+			_marketParams, // Updated token contracts in case of Eth
+			_amt // Shares amount converted to assets
+		) = _performEthToWethSharesConversion(
+			_marketParams,
+			_shares,
+			_onBehalf,
+			_getId,
+			false
+		);
+
+		approve(
+			TokenInterface(_marketParams.loanToken),
+			address(MORPHO_BLUE),
+			_amt
+		);
+
+		(uint256 _assets, ) = MORPHO_BLUE.supply(
+			_marketParams,
+			_amt,
```

Comment:
Should be set to 0 right?

Reply:
You mean the `_amt` value? So I've converted the shares value to assets amount to give the approval to Morpho's contract. That's why sending the converted assets value in the function.